pkg/server/registry/apigroups/acorn/apps/validator.go

```diff
@@ -584,13 +584,20 @@ func (s *Validator) checkScheduling(ctx context.Context, params *apiv1.App, proj
 		return append(validationErrors, field.Invalid(field.NewPath("spec", "image"), params.Spec.Image, fmt.Sprintf("error listing compute classes: %v", err)))
 	}
 
+	filteredComputeClasses := new(apiv1.ComputeClassList)
+	for _, cc := range computeClasses.Items {
+		if slices.Contains(cc.SupportedRegions, params.Spec.Region) || slices.Contains(cc.SupportedRegions, defaultRegion) {
```

Comment:
If the `params.Spec.Region != ""`, then we don't want to check the `defaultRegion` here.

However, we also need to collect all the compute classes that may be specified by a user or acornfile. In order of precidence:
1. `params.Spec.ComputeClasses`
2. `app.Status.AppSpec.{Containers,Jobs}[].ComputeClass` as well as `app.Status.AppSpec.{Containers,Jobs}[].Sidecars[].ComputeClass`
3. The default for the region.

By including all of these here, the checks below will catch if an app is trying to use a compute class that is not for the region.

In order to check 2, you would need the `imageDetails.AppSpec` from where this method is called.

Reply:
➕